BridgeSDK/BridgeAPI/SBBUploadManager.m

```diff
@@ -544,42 +547,109 @@ - (void)uploadFileToBridge:(NSURL *)fileUrl contentType:(NSString *)contentType
 - (void)kickOffUploadForFile:(NSString *)filePath uploadRequestJSON:(NSDictionary *)uploadRequestJSON
 {
     // this starts the process by downloading the Bridge UploadSession.
+    // first, make sure we get rid of any existing UploadSession for this file, in case this is a retry.
+    [self setUploadSessionJSON:nil forFile:filePath];
```

Comment:
it's effectively synchronized by only being called from within the background session delegate queue. I remove the upload session here so the completion handler doesn't see it's already got one and ignore it as a duplicate session due to the retry crossing paths with the response finally coming in from the original request that was being retried.